core/commands/partials/terraform_configurator.rb

```diff
@@ -23,44 +24,47 @@ def initialize(config, env, logger)
                         else
                           NetworkSettings.new
                         end
+    @threads_count = @env.threads_count
+    Workers.pool.resize(@threads_count)
   end
 
   # Brings up nodes
   #
-  # @return [Number] execution status
+  # @return [Result::Base] execution result
   def up
     nodes = @config.node_names
-    up_results = nodes.map { |node| bring_up_and_configure(node) }
-    return Result.error('') unless up_results.detect(&:!).nil?
-
-    Result.ok('')
+    up_machines(nodes).and_then do
+      configure_machines(nodes)
+    end
   end
 
   private
 
   # Check whether chef have provisioned the server or not
   #
   # @param node [String] name of the node to check
-  # return [Boolean]
-  def node_provisioned?(node)
+  # @param logger [Out] logger to log information to
+  # @return [Boolean]
+  def node_provisioned?(node, logger)
     node_settings = @network_settings.node_settings(node)
     result = TerraformService.ssh_command(node_settings,
```

Comment:
Use MachineConfigurator here instead of the TerraformService. The proxy is not needed here.